File: app/src/main/java/netpro/keyTransmitter/SpaceItemDecoration.java
package netpro.keyTransmitter;

import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * RecyclerViewの子View間のスペースを設定するクラスです
 */
public class SpaceItemDecoration extends RecyclerView.ItemDecoration {

    private final int topSpace;
    private final int bottomSpace;
    private final int rightSpace;
    private final int leftSpace;

    public SpaceItemDecoration(int topSpace, int spaceBottom, int spaceRight, int leftSpace) {
        this.topSpace = topSpace;
        this.bottomSpace = spaceBottom;
        this.rightSpace = spaceRight;
        this.leftSpace = leftSpace;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.set(leftSpace, topSpace, rightSpace, bottomSpace);
    }
}

File: app/src/main/java/jp/gr/java_conf/snake0394/keytransmitter/SpaceItemDecoration.kt
package jp.gr.java_conf.snake0394.keytransmitter

import android.graphics.Rect
import android.support.v7.widget.RecyclerView
import android.view.View

/**
 * RecyclerViewの子View間のスペースを設定するクラス
 */
class SpaceItemDecoration(private val topSpace: Int, private val bottomSpace: Int, private val rightSpace: Int, private val leftSpace: Int) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State?) {
        outRect.set(leftSpace, topSpace, rightSpace, bottomSpace)
    }
}